File: src/Truck.cpp
//
// Created by Hazhir on 12/22/2023.
//

#include "Truck.h"

#include <utility>

Truck::~Truck() = default;

int Truck::getId() const {
    return id;
}


void Truck::setDirection(const std::string &newDirection) {
    Truck::direction = newDirection;
}

void Truck::setSpeed(float newSpeed) {
    Truck::speed = newSpeed;
}

void Truck::setDestination(const GPSLocation &newDestination) {
    Truck::destination = newDestination;
}

GPSLocation Truck::getLocation() {
    return location;
}

void Truck::setLocation(const GPSLocation &newLocation) {
    Truck::location = newLocation;
}

Truck::Truck(int id, std::string direction, float speed, GPSLocation &destination,
             GPSLocation &location) : id(id), direction(std::move(direction)), speed(speed), destination(destination),
                                            location(location) {
    this->collisionSensor = *new CollisionSensor();
}

std::string Truck::getDirection() const {
    return direction;
}

float Truck::getSpeed() const {
    return speed;
}

GPSLocation Truck::getDestination() const {
    return destination;
}

void Truck::changeSpeed(float newSpeed) {
    this->speed = newSpeed;
}

void Truck::changeDestination(GPSLocation &newDestination) {
    this->destination = newDestination;
}

void Truck::changeDirection(std::string &newDirection) {
    Truck::direction = newDirection;
}

void Truck::stop() {
    printf("stopping");
    setDirection("right");
    setSpeed(10);
    setDirection("straight");
    setSpeed(0);
}

void Truck::operator()() {

}

int Truck::getMessageCounter() const {
    return messageCounter;
}

void Truck::setMessageCounter(int messageCounter1) {
    Truck::messageCounter = messageCounter1;
}

int Truck::incrementAndGetMessageCounter() {
    return ++Truck::messageCounter;
}

File: src/src/Truck.java
package src;

public class Truck {
    private String direction;
    private double speed;
    private double maintainDistance;
   // private GPSLocation destination;
    //private GPSLocation location;
    private boolean collisionSensor;

    private String ipAddress;
    private int port;
    

    public Truck(String ipAddress, int port) {
        this.ipAddress = ipAddress;
        this.port = port;
    }


	public String getDirection() {
		return direction;
	}


	public void setDirection(String direction) {
		this.direction = direction;
	}


	public double getSpeed() {
		return speed;
	}


	public void setSpeed(double speed) {
		this.speed = speed;
	}


	public double getMaintainDistance() {
		return maintainDistance;
	}


	public void setMaintainDistance(double maintainDistance) {
		this.maintainDistance = maintainDistance;
	}


	public boolean isCollisionSensor() {
		return collisionSensor;
	}


	public void setCollisionSensor(boolean collisionSensor) {
		this.collisionSensor = collisionSensor;
	}


	public String getIpAddress() {
		return ipAddress;
	}


	public void setIpAddress(String ipAddress) {
		this.ipAddress = ipAddress;
	}


	public int getPort() {
		return port;
	}


	public void setPort(int port) {
		this.port = port;
	}
    

    // Getters and setters for all fields
    // Additional methods like changeSpeed, changeDirection, etc.
}
